### PR TITLE
🎇Story Ads: Development mode

### PR DESCRIPTION
**Pull request by calebcordry**
Introduces new `development` mode for amp-story-auto-ads if you include a `development` attribute on the tag.

```html
<amp-story-auto-ads id="i-amphtml-demo-1" development>
    ...
</amp-story-auto-ads>
```

Really helpful for iterative ad development, and will be used in story preview tool.

### CODE REVIEW

File: extensions/amp-story-auto-ads/0.1/test/test-amp-story-auto-ads.js [hunk 1]
```diff
@@ -396,5 +399,54 @@ describes.realWin(
...
+        const dispatchStub = sandbox.spy(storyEvents, 'dispatch');
+
+        const ampAd = doc.querySelector('amp-ad');
+        ampAd.signals().signal(CommonSignals.INI_LOAD);
+        await macroTask();
+
+        expect(insertSpy).calledWith('story-page-0', 'i-amphtml-ad-page-1');
+        const payload = {
+          'targetPageId': 'i-amphtml-ad-page-1',
+          'direction': NavigationDirection.NEXT,
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/test/test-amp-story-auto-ads.js]: glad you have this! that makes sure a test failure when the constant is changed.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 2]
```diff
@@ -223,14 +227,31 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
   forceRender(pageBeforeAdId) {
+    const pageBeforeId =
+      pageBeforeAdId ||
+      /** @type {string} */ (this.storeService_.get(
+        StateProperty.CURRENT_PAGE_ID
+      ));
     this.isCurrentAdLoaded_ = true;
-    this.tryToPlaceAdAfterPage_(pageBeforeAdId);
+    // Setting distance manually to avoid flash of next page.
+    const lastPage = this.adPageEls_[this.adPageEls_.length - 1];
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: worth to extract a method and shared in array.js

`function lastItem(array)`

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 3]
```diff
@@ -223,14 +227,31 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
+    lastPage.setAttribute('distance', '1');
+    this.tryToPlaceAdAfterPage_(pageBeforeId);
+    // Once the ad is inserted into the story firing this event will
+    // navigate to the ad page.
+    const payload = dict({
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: extracting method is a better way to document the behavior.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 4]
```diff
@@ -223,14 +227,31 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
   }
 
   /**
-   * Force this extension to behave as if ad is about to be shown.
-   * Used for testing but should be extended to force an ad to show in a real story.
-   * @param {string} pageBeforeAdId
+   * Force this extension to render all amp-story-auto-ads controlled UI and
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Force an immediate ad placement without waiting for ad being loaded, and then navigate to the ad page.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 5]
```diff
@@ -223,14 +227,31 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
+   * navigate to ad.
+   * @param {string=} pageBeforeAdId
    * @visibleForTesting
    */
   forceRender(pageBeforeAdId) {
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: `forceRender` is ambiguous. how about `forcePlaceAdAfterPage`

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js (hunk 4 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Done.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js (hunk 2 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Done.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js (hunk 5 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Done.

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js (hunk 3 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Done.

Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Done.

**Pull request merged**